### PR TITLE
collector: upgrade collector core components to v0.102

### PR DESCRIPTION
**Pull request by ridwanmsharif**
Change-Id: Id11d485e02a1393181e64d6eb9e9e508098475ce

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.20.12 as builder
+FROM golang:1.21.0 as builder
```
Comment (braydonk) [inline Dockerfile:1]: Any reason not to go to 1.21.11?

File: Makefile [hunk 2]
```diff
@@ -63,7 +63,9 @@ install-tools:
 			github.com/client9/misspell/cmd/misspell \
 			github.com/golangci/golangci-lint/cmd/golangci-lint \
 			github.com/google/addlicense \
-			github.com/open-telemetry/opentelemetry-collector-contrib/cmd/mdatagen
+			go.opentelemetry.io/collector/cmd/mdatagen \
+			github.com/google/googet/goopack \
```
Comment (braydonk) [inline Makefile:67]: Do you do a googet package of this repo? If not you probably don't need this

**Review by braydonk — APPROVED** [review]
Two small comments but LGTM otherwise

File: Dockerfile (hunk 1 above)
Comment (ridwanmsharif) [reply Dockerfile]: done

File: Makefile (hunk 2 above)
Comment (ridwanmsharif) [reply Makefile]: done

**Pull request merged**